run_project_tests.py

```diff
@@ -155,39 +156,22 @@ def stop_handler(signal, frame):
 # unity_flags = ['--unity']
 unity_flags = []
 
-backend_flags = None
-compile_commands = None
-test_commands = None
-install_commands = []
-clean_commands = []
-
 def setup_commands(backend):
-    global backend_flags, compile_commands, test_commands, install_commands, clean_commands
+    global do_debug, backend_flags
+    global compile_commands, clean_commands, test_commands, install_commands, uninstall_commands
     msbuild_exe = shutil.which('msbuild')
     if (backend and backend.startswith('vs')) or (backend is None and msbuild_exe is not None):
         if backend is None:
-            backend = 'vs2010'
+            # Autodetect if unspecified
+            backend = 'vs'
         backend_flags = ['--backend=' + backend]
-        compile_commands = ['msbuild']
-        test_commands = ['msbuild', 'RUN_TESTS.vcxproj']
     elif backend == 'xcode' or (backend is None and mesonlib.is_osx()):
         backend_flags = ['--backend=xcode']
-        compile_commands = ['xcodebuild']
-        test_commands = ['xcodebuild', '-target', 'RUN_TESTS']
     else:
+        backend = 'ninja'
         backend_flags = []
-        # We need at least 1.6 because of -w dupbuild=err
-        ninja_command = environment.detect_ninja(version='1.6')
-        if ninja_command is None:
-            raise RuntimeError('Could not find Ninja v1.6 or newer')
-        if do_debug:
-            compile_commands = [ninja_command, '-v']
-        else:
-            compile_commands = [ninja_command]
-        compile_commands += ['-w', 'dupbuild=err']
-        test_commands = [ninja_command, 'test', 'benchmark']
-        install_commands = [ninja_command, 'install']
-        clean_commands = [ninja_command, 'clean']
+    compile_commands, clean_commands, test_commands, install_commands, \
+            uninstall_commands = get_backend_commands(backend, do_debug)
```

Comment:
__[flake8]__
* [E127] continuation line over-indented for visual indent

<!-- Comment by [SideCI](https://sideci.com) -->